src/states_screens/race_result_gui.cpp

```diff
@@ -1457,6 +1459,38 @@ void RaceResultGUI::unload()
                 NULL, true /* ignoreRTL */);
         }
 
+        // Draw the difficulty
+        // -------------------
+        core::stringw difficulty_one;
+        core::stringw difficulty_two;
+        core::recti diff_ghost = core::recti(current_x, y, current_x + 200, y + 20);
+
+        if (RaceManager::get()->hasGhostKarts() && ReplayPlay::get()->isSecondReplayEnabled())
+        {
+            unsigned idw = ReplayPlay::get()->getCurrentReplayFileIndex();
+            unsigned idx = ReplayPlay::get()->getSecondReplayFileIndex();
+            const ReplayPlay::ReplayData& rd1 = ReplayPlay::get()->getReplayData(idw);
+            const ReplayPlay::ReplayData& rd2 = ReplayPlay::get()->getReplayData(idx);
+            difficulty_one = RaceManager::get()->getDifficultyName((RaceManager::Difficulty)rd1.m_difficulty);
+            difficulty_two = RaceManager::get()->getDifficultyName((RaceManager::Difficulty)rd2.m_difficulty);
+
+            if (m_second_ghost == false) {
+                if (rd1.m_difficulty > rd2.m_difficulty){
+                    m_font->draw(difficulty_one, diff_ghost, color, false, false, NULL, true);
+                } else {
+                    m_font->draw(difficulty_two, diff_ghost, color, false, false, NULL, true);
+                }
+                m_second_ghost = true;
+            } else {
+                if (rd1.m_difficulty > rd2.m_difficulty){
+                    m_font->draw(difficulty_two, diff_ghost, color, false, false, NULL, true);
+                } else {
+                    m_font->draw(difficulty_one, diff_ghost, color, false, false, NULL, true);
+                }
+                m_second_ghost = false;
+            }
```

Comment:
Maybe, this would be better (less duplication) (Have not tested)

```cpp
        if (RaceManager::get()->hasGhostKarts() && ReplayPlay::get()->isSecondReplayEnabled())
        {
            const unsigned idw = ReplayPlay::get()->getCurrentReplayFileIndex();
            const unsigned idx = ReplayPlay::get()->getSecondReplayFileIndex();
            const ReplayPlay::ReplayData& rd1 = ReplayPlay::get()->getReplayData(idw);
            const ReplayPlay::ReplayData& rd2 = ReplayPlay::get()->getReplayData(idx);
            const core::stringw difficulty_one = RaceManager::get()->getDifficultyName((RaceManager::Difficulty)rd1.m_difficulty);
            const core::stringw difficulty_two = RaceManager::get()->getDifficultyName((RaceManager::Difficulty)rd2.m_difficulty);
            const core::recti diff_ghost = core::recti(current_x, y, current_x + 200, y + 20);
            
            core::stringw difficulty = difficulty_two;
            if(!m_second_ghost && rd1.m_difficulty > rd2.m_difficulty ||
                m_second_ghost && rd1.m_difficulty <= rd2.m_difficulty)
                difficulty = difficulty_one;

            m_font->draw(difficulty, diff_ghost, color, false, false, nullptr, true);
            m_second_ghost = !m_second_ghost;
        }
```